schema.go

```diff
@@ -92,3 +92,8 @@ var (
 // http://json-schema.org/latest/json-schema-validation.html#rfc.section.5.26
 // RFC draft-wright-json-schema-validation-00, section 5.26
 type Definitions map[string]*Schema
+
+// Type implements section 6.1.1
+type Type struct {
+	Types []string
+}
```

Comment:
Did you already consider this?

```suggestion
type Type []string
```

I'd also suggest adding helper methods to avoid all the messy assignments, instead of:
```
&Type{Types: []string{"string"}}
```
I'd expect:
```
MakeType("string")
```

With a method pattern that looks like:

```
func MakeType(typ ...string) Type
```

(I'm using `Make` in the method name as opposed to `New` reflecting the underlying slice instead of an object.)